addok/shell.py

```diff
@@ -49,7 +49,11 @@ def _init_history_file(self):
             atexit.register(self.save_history)
 
     def save_history(self):
-        readline.write_history_file(self.history_file)
+        try:
+            readline.write_history_file(self.history_file)
+        except FileNotFoundError:
+            print(red('Unable to write history file to '
```

Comment:
Because it's a bit tricky to distinguish the case of FileNotFoundError because the log dir does not exist from the case where the file itself does not exist (which is normal at first shell run).